### PR TITLE
Private events

### PR DESCRIPTION
**Pull request by uryelah**
This is an events site project, similar to Eventbrite, done with ruby on rails that allows users to create events, send invites to their hosted or attending events and confirm their presence for the events they were invited to. 
- Sign up and log in to access the events, invites and users routes.
- Create an event and send invites to other users
- Receive and accept an invite to an event to be able to attend it and send invites for it
- Cancel your attendance to an event to destroy your invitation
- An event invite cannot be sent to its creator
- A user cannot send an invite to an event held past the current time

UPDATE:
- required code reviews changes applied

UPDATE2:
- fix bug when for existing user cookie but no correspondent user on db
- add unit tests to the models
- add integration tests to auth paths
- add integration tests for creation and inviting of/to events

### CODE REVIEW

**Review by Redvanisation — CHANGES_REQUESTED** [review]
# Changes required  ♻️ 

### Hello team  🌟

**Very nice work on this project, however, the following issues need to be fixed before moving on ⬇️** 

1. Please implement the following in the **UsersController `show` action** and not in the **views' helpers**:

* [x] lists events created by user, i.e. `@created_events = current_user.events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_show_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [x] lists events previoulsy attendented by user **using event model scope**, i.e. `@past_events = current_user.attended_events.previous` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [x] lists upcoming events previously attended by user **using event model scope**, i.e. `@upcoming_events = current_user.attended_events.upcoming` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations)).

2. * [x] There should be NO application logic in the views nor helpers (no `Post.new`, `Comment.create`, `current_user.friends.find_by(...)`...). Everything connected with data processing should be encapsulated in models and used in controllers, [check](http://codefol.io/posts/Where-Do-I-Put-My-Code) this for reference, [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png).

3. The tests are **red** please make them **green**:
Screen Shot 2019-12-26 at 1 38 15 PM

### _After making the required changes please submit a new code review request_

### Good luck! 🍀

**Review by Redvanisation — CHANGES_REQUESTED** [review]
# Changes required  ♻️ 

### Hello team  🌟

**Great work on this project, however, the following issues need to be fixed before moving on ⬇️** 

1. I tried to access the project but I'm getting the following error: 

Screen Shot 2019-12-27 at 7 06 37 PM

2. **Automated Tests:**
* [x] The models should have unit tests for validations and associations **(Rspec)** ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/unit_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/)).
* [x] The authentication needs to have integration tests **(Rspec + Capybara)** ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/)).
* [x] Event managment features need have integration tests **(Rspec + Capybara)** ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/)).

- You could refer to my [article on here](https://hackernoon.com/test-driven-development-with-rspec-in-rails-gsr3zrs) about testing if you don't know how to implement Rspec and Capybara.

### _After making the required changes please submit a new code review request_

### Good luck! 🍀

File: spec/models/event_spec.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    it 'Test if event with invalid creator id is invalid' do
+    	expect(event2.save).to be false
+		end
+		
+		it 'Test if event without title is invalid' do
+			event3.title = ''
+			expect(event3.valid?).to be false
+			event3.title = 'title'
+		end
+
```
Comment (samgaco) [inline spec/models/event_spec.rb:24]: :medal_military:

**Review by samgaco — APPROVED** [review]
Hi @uryelah and @jairjy :high_brightness: 

The error indicated in the previous review is not showing anymore and integration & unit test were added. You can move on :ballot_box_with_check: 

Well done with the tests by the way :+1: 

Keep up the good work! 

------------------

* ###### STATUS OF THE PROJECT: APPROVED :heavy_check_mark:

**Pull request merged**